app/src/main/java/com/example/instasnap/HomePage.java

```diff
@@ -0,0 +1,65 @@
+package com.example.instasnap;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.TextView;
+
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+
+public class HomePage extends AppCompatActivity {
+
+    private FirebaseAuth _auth;
+    private Button _logoutButton;
+    private TextView _welcomeText;
+    private FirebaseUser _user;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_homepage);
+
+        initialize();
+
+        checkLoginStatus();
+
+        setLogoutButtonListener();
+
+    }
+
+
+    private void initialize() {
+        _auth = FirebaseAuth.getInstance();
+        _logoutButton = findViewById(R.id.logout_button);
+        _welcomeText = findViewById(R.id.welcome_textview);
+        _user = _auth.getCurrentUser();
+    }
+
+    private void setLogoutButtonListener() {
+        _logoutButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                FirebaseAuth.getInstance().signOut();
+                Intent loginIntent = new Intent(getApplicationContext(), Login.class);
+                startActivity(loginIntent);
+                finish();
+            }
+        });
+    }
+
+    private void checkLoginStatus() {
```

Comment:
Is it possible to be in the HomePage activity when the user is null ??

Reply:
At the moment yea but it will be fixed

Reply:
Can you explain how? Maybe we should have a "landing" class that routes to the correct activity.

Reply:
If i delete the user on the FB, he'll still be considered logged in

Reply:
I think we can delete this validation, your choice